### DISCUSSION TITLE
3.1.0-rc.1

### DISCUSSION THREAD

**Discussion by janusw** [dotnet-bluetooth-le/dotnet-bluetooth-le#842] [Announcements]
* #820 Windows Implementation for BondAsync and BondedDevices
* #839 Windows: Handle that a device is already connected

<hr /><em>This discussion was created from the release <a href='https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/releases/tag/3.1.0-rc.1'>3.1.0-rc.1</a>.</em>

**Comment by axa88** [comment]
bit the bullet and updated to this version...
Starting to see some odd behavior.

here it seems that when the device had lost connection/disconnected, a call to Get the BondState was made...
Apparently the underlying `NativeDevice` that _once was connected_ was perhaps disposed of and the the implementation throws... 

I suppose this isn't new behavior but easily missed if not running the debugger.

![image](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/assets/13371986/5710d6ac-557a-4e4f-b7b9-fc9db1ddc182)

Apparently when the device has been disposed of when disconnected the state is actually unavailable.
So perhaps return an additional state "Unavailable", as it gives too many undefined meanings to `NotSupported`.

**Reply by janusw** [reply]
> bit the bullet and updated to this version... Starting to see some odd behavior.

I don't fully understand what you're doing and how it misbehaves. If you think there is a bug, please open a proper bugreport.

**Comment by axa88** [comment]
> I don't fully understand what you're doing and how it misbehaves. If you think there is a bug, please open a proper bugreport.


This issue here has already been filed as #850 .  That issue reports a call to `Adapter.DisconnectDeviceAsync` not completing, and when that happens, the associated `IDevice` remains in the list of `Adapter.ConnectedDevices` (even as it reports as `State = Disconnected`).

Then when probing for the `IDevice.Bondstate`, the call to `GetBondState()` fails as depicted above because `IDevice.NativeDevice` is null (perhaps due to the disconnection??).

![image](https://github.com/dotnet-bluetooth-le/dotnet-bluetooth-le/assets/13371986/dcc17b9f-8633-40ae-8a19-406ae1b0d22c)


So basically this issue reported is looking to me just one manifestation of bug #850.
Another manifestation could be as stated above, the Disconnected Device remains in `Adapter.ConnectedDevices` list, or that `IDevice.NativeDevice` is null... etc...
